config.example.wikimedia.yaml

```diff
@@ -450,6 +443,170 @@ templates:
                 request:
                   uri: https://wikimedia.org/api/rest_v1/metrics/pageviews/top/{+rest}
 
+      /{module:graphoid}/v1/store/{format}:
+        post:
+          x-setup-handler:
+            - init:
+                uri: /{domain}/sys/post_data/graphoid.request
+            - init:
+                uri: /{domain}/sys/key_value/graphoid.png
+                body:
+                  keyType: string
+                  valueType: blob
+            - init:
+                uri: /{domain}/sys/key_value/graphoid.svg
+                body:
+                  keyType: string
+                  valueType: blob
+          x-request-handler:
+            - calculate_hash:
+                request:
+                  method: put
+                  uri: /{domain}/sys/post_data/graphoid.request/hash
+                  body: '{$.request.body}'
+            - check_storage:
+                request:
+                  method: get
+                  uri: /{domain}/sys/key_value/graphoid.{format}/{$.calculate_hash.body}
+                  headers:
+                    cache-control: '{cache-control}'
+                return_if:
+                  status: '2xx'
+                return:
+                  status: 200
+                  headers: '{$.check_storage.headers}'
+                  body: '{$.check_storage.body}'
+                catch:
+                  status: 404
+            - load_data:
+                request:
+                  method: post
+                  uri: /{domain}/sys/graphoid/v1/handler/{format}
+                  headers:
+                    cache-control: '{cache-control}'
+                    x-resource-location: '{$.calculate_hash.body}'
+                  body: '{$.request.body}'
+            - store_request:
+                request:
+                  method: put
+                  uri: /{domain}/sys/post_data/graphoid.request/
+                  body: '{$.request.body}'
+            - return_content:
+                return:
+                  status: 200
+                  headers: '{$.load_data.headers}'
+                  body: '{$.load_data.body}'
+      /{module:graphoid}/v1/preview/{format}:
+        post:
+          x-request-handler:
+            - calculate_hash:
+                request:
+                  method: put
+                  uri: /{domain}/sys/post_data/graphoid.request/hash
+                  body: '{$$.merge($.request.body, {"x-restbase-mode": "preview"})}'
+            - handle_req:
+                request:
+                  method: post
+                  uri: /{domain}/sys/graphoid/v1/handler/{$.request.params.format}
+                  headers:
+                    max-age: 3600
```

Comment:
It would be nice to use a shared stash for this, instead of relying on per-row TTLs.